### PR TITLE
fix(internal): more miscellaneous bento fixes

### PR DESCRIPTION
**Pull request by sauyon**

### CODE REVIEW

File: bentoml/_internal/bento/bento.py [hunk 1]
```diff
@@ -279,8 +265,11 @@ def from_yaml_file(cls, stream: t.IO[t.Any]):
             logger.error(exc)
             raise
 
+        yaml_content["tag"] = Tag(yaml_content["name"], yaml_content["version"])
+        del yaml_content["name"]
+        del yaml_content["version"]
+        yaml_content["creation_time"] = datetime.fromisoformat(yaml_content["creation_time"])
         bento_info = cls(**yaml_content)
```
Comment (parano) [inline bentoml/_internal/bento/bento.py:272]: should we catch the `TypeError` for unexpected keyword argument here and give user a message about unexpected content in `bento.yaml` ?

File: bentoml/_internal/bento/bento.py [hunk 2]
```diff
@@ -144,13 +143,14 @@ def create(
...
+            BentoInfo(tag, svc, labels, models).dump(bento_yaml)  # type: ignore
 
-        return OSBento(tag, bento_fs)
+        return SysPathBento(tag, bento_fs)
 
     @classmethod
-    def from_fs(cls, tag: Tag, fs: FS) -> "Bento":
-        res = cls(Tag("", ""), fs)
+    def from_fs(cls, item_fs: FS) -> "Bento":
+        res = cls(None, item_fs)  # type: ignore
```
Comment (aarnphm) [inline bentoml/_internal/bento/bento.py:153]:
```suggestion
        res = cls(None, item_fs)  # type: Bento
```

File: bentoml/_internal/bento/bento.py [hunk 3]
```diff
@@ -277,8 +264,23 @@ def from_yaml_file(cls, stream: t.IO[t.Any]):
             logger.error(exc)
             raise
 
-        bento_info = cls(**yaml_content)
-        bento_info.validate()
+        yaml_content["tag"] = Tag(yaml_content["name"], yaml_content["version"])
+        del yaml_content["name"]
```
Comment (aarnphm) [inline bentoml/_internal/bento/bento.py:268]: Why do we have to do this?

File: bentoml/_internal/bento/bento.py [hunk 4]
```diff
@@ -234,41 +236,26 @@ def save(
...
-            labels=labels,  # TODO: validate user provided labels
-            apis=svc._apis,  # type: ignore[reportPrivateUsage]
-            models=models,  # TODO: populate with model & framework info
-        )
-
-        bento_info.validate()
-        return bento_info
-
     @classmethod
     def from_yaml_file(cls, stream: t.IO[t.Any]):
```
Comment (aarnphm) [inline bentoml/_internal/bento/bento.py:260]: maybe it is type `BinaryIO` instead?

```suggestion
    def from_yaml_file(cls, stream: t.BinaryIO):
```

File: bentoml/_internal/bento/bento.py (hunk 2 above)
Comment (sauyon) [reply bentoml/_internal/bento/bento.py:153]: The issue is the `None` tag argument, not the result type of the expression.

File: bentoml/_internal/bento/bento.py (hunk 3 above)
Comment (sauyon) [reply bentoml/_internal/bento/bento.py:268]: I'm passing the yaml directly as kwargs to init.

File: bentoml/_internal/bento/bento.py (hunk 4 above)
Comment (sauyon) [reply bentoml/_internal/bento/bento.py:260]: Why? Does the method not support `TextIO`?

Comment (aarnphm) [reply bentoml/_internal/bento/bento.py:260]: Does passing a file in directly considered a TextIO? Passing files would be considered as a BinaryIO no?

Comment (sauyon) [reply bentoml/_internal/bento/bento.py:260]: If you open a file with an encoding it's `TextIO`, I believe.

**Comment by sauyon** [comment]
Oh the test failures are because of the `DummyItem` class...

**Comment by sauyon** [comment]
@aarnphm @parano  seems like the e2e tests are failing but this should be good to go in---codestyle checks are running on everything as I said, so it's not very useful.

**Comment by sauyon** [comment]
...nevermind I introduced an `isort` issue while cleaning up `pylint` issues, of course.

File: bentoml/_internal/bento/bento.py (hunk 2 above)
Comment (bojiang) [reply bentoml/_internal/bento/bento.py:154]: why we cannot make this class `tag: t.Optiional[Tag]` here as usual?

**Comment by parano** [comment]
@sauyon I think the issue `No module named pytest` in test CI has been fixed, you can try re-run tests on this PR now

**Pull request merged**